dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -76,8 +75,8 @@ public static Fence newFenceOpt(String name, String opt) {
     	return fence;
     }
 
-    public static Init newInit(IExpr address, IConst value) {
-        return new Init(address, value);
+    public static Init newInit(Address base, int offset) {
```

Comment:
`Init` now has a reference to the initial value of the location, instead of a copy.  See `#getValue()`.

Reply:
I agree with this change. An address should not care about the value it points to initially, just like it doesn't care how the values evolve over the execution. The `init`-events are what give initial values to addresses instead.

Reply:
I see `Init` more as a necessity to the modelling language, than as an integral part of the program.  At some point we might want to be lazy with the creation of `Init`, considering that `AtomicAsLock` already adds one post-build, and the planned escape analysis would allow to remove some.